### PR TITLE
Extract jvm version from openjdk vulnerabilities group

### PR DESCRIPTION
**Pull request by Scanteianu**
Two steps required here:

1) Extract versions mentioned in the report (already done).
2) intersect list of all versions with major versions reported in the dot matrix for each cve..

Also includes unit test  coverage for this version extraction. Disables the NIST version extraction.

### CODE REVIEW

File: cvereporter/fetch_vulnerabilities.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def retrieve_cves_from_internet(date: str) -> str:
                 "Referer": "http://www.google.com/",
             },
         )
-        print(r)
+        # print(r)
```
Comment (karianna) [inline cvereporter/fetch_vulnerabilities.py:50]: Remove commented out code?

File: cvereporter/fetch_vulnerabilities.py [hunk 2]
```diff
@@ -92,27 +92,45 @@ def parse_to_dict(resp_text: str, date: str) -> list[dict]:
                 while score.find_next_sibling("th") is not None:
                     versions.append(score.find_next_sibling("th").text)
                     score = score.find_next_sibling("th")
+            # extract table column headers
+            if "CVE ID" in header.text:
```
Comment (karianna) [inline cvereporter/fetch_vulnerabilities.py:96]: extract to its own function?

File: cvereporter/fetch_vulnerabilities.py [hunk 3]
```diff
@@ -92,27 +92,45 @@ def parse_to_dict(resp_text: str, date: str) -> list[dict]:
...
-            versionCheck = scoreTD
             affected_versions = []
-            affected_versions += (
-                extracted_affected  # todo - maybe just the extracted ones
-            )
-            for version in versions:
-                versionCheck = versionCheck.find_next_sibling("td")
-                if versionCheck.text == "•":
-                    affected_versions.append(int(version))
+            for version in extracted_affected:
```
Comment (karianna) [inline cvereporter/fetch_vulnerabilities.py:120]: Would be good to have a comment showing some sample data.

File: cvereporter/nist_enhance.py [hunk 4]
```diff
@@ -102,7 +102,10 @@ def enhance(vulns: list[Vulnerability]):
             )
             vuln.ratings.add(vr)
         vuln.description = relevant["description"]
-        for affects in vuln.affects:
-            for ver in relevant["versions"]:
-                affects.versions.add(ver)
+        # for now - we use ojvg's version extraction
```
Comment (karianna) [inline cvereporter/nist_enhance.py:105]: Maybe expand on what ojvg is

File: tests/test_pipeline.py [hunk 5]
```diff
@@ -1,26 +1,41 @@
 from cvereporter import fetch_vulnerabilities, nist_enhance
 import json
 
+
+# to run just one test: python3 -m pytest -v -k test_fetch -s (in this case, runs "test_fetch")
```
Comment (karianna) [inline tests/test_pipeline.py:5]:
```suggestion
# To run a single test: python3 -m pytest -v -k test_fetch -s (in this case, runs "test_fetch")
```

**Review by karianna — CHANGES_REQUESTED** [review]

**Comment by Scanteianu** [comment]
thanks @karianna for the review, i think i addressed everything

**Review by karianna — APPROVED** [review]

**Review by smlambert — APPROVED** [review]

**Pull request merged**